### PR TITLE
Adding WindowsVersion to Get-ComputerInfo

### PR DESCRIPTION
**Pull request by mirichmo**
Changes made:
1. Added WindowsVersion to Get-ComputerInfo
2. Ported the internal tests for Get-ComputerInfo to Pester for PowerShell Core
3. Added a test for WindowsVersion

### CODE REVIEW

**Comment by mirichmo** [comment]
@jeffbi and @ChrisUbben Please take a look

**Comment by jeffbi** [comment]
LGTM

File: test/powershell/Modules/Microsoft.PowerShell.Management/Get-ComputerInfo.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,1374 @@
+# This is a Pester test suite to validate the cmdlets in the ComputerInfo module
+#
+# Copyright (c) Microsoft Corporation, 2016
```
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Get-ComputerInfo.Tests.ps1]: Copyright is not needed here.

Comment (mirichmo) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Get-ComputerInfo.Tests.ps1]: Fixed

File: test/powershell/Modules/Microsoft.PowerShell.Management/Get-ComputerInfo.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,1371 @@
...
+
+        ## HELPER METHODS RELYING ON P-INVOKE BEGIN ******************************************
+    function Get-BiosFirmwareType
+    {
+        [int]$firmwareType = 0
+        (Get-FirmwareType)::GetFirmwareType([ref]$firmwareType)
+        return $firmwareType
+    }
+
+
```
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Get-ComputerInfo.Tests.ps1]: There are inconsistencies in number of new lines between functions. Some use 1 line space, some use 2, etc.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Get-ComputerInfo.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,1371 @@
...
+              
+            "HyperVisorPresent" {return Get-HyperVProperty $propertyName}
+            "HyperVRequirementDataExecutionPreventionAvailable" {return Get-HyperVProperty $propertyName}
+            "HyperVRequirementSecondLevelAddressTranslation" {return Get-HyperVProperty $propertyName}
+            "HyperVRequirementVirtualizationFirmwareEnabled" {return Get-HyperVProperty $propertyName}
+            "HyperVRequirementVMMonitorModeExtensions" {return Get-HyperVProperty $propertyName}
+            "KeyboardLayout" {return Get-KeyboardLayout}
+            "LogonServer" {return [Microsoft.Win32.Registry]::GetValue("HKEY_Current_User\Volatile Environment", "LOGONSERVER", "")}
+             
+
```
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Get-ComputerInfo.Tests.ps1]: Same line space inconsistencies here.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Get-ComputerInfo.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,1371 @@
...
+            "PropertyName" = $propertyName}
+        $testCases += $testCase 
+    }
+    $testCases      
+}
+
+function Get-CommonProperties
+{
+    param($observed,$expected)
+
```
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Get-ComputerInfo.Tests.ps1]: New line inconsistency again. Some use no new line after param, some use 1 line, etc.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Get-ComputerInfo.Tests.ps1 [hunk 5]
```diff
@@ -0,0 +1,1371 @@
...
+    Describe "Tests for Get-ComputerInfo: Ensure Type returned" -tags "CI", "RequireAdminOnWindows" {
+        It "Verfiy type returned by Get-ComputerInfo" {
+            $computerInfo = Get-ComputerInfo
+            $computerInfo.GetType().Name | Should Be "ComputerInfo"
+        } 
+    }
+
+    Describe "Tests for Get-ComputerInfo" -tags "Feature", "RequireAdminOnWindows" {
+
```
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Get-ComputerInfo.Tests.ps1]: New line inconsistency. Some describes have no space after, some have 1 line, etc.

**Comment by mirichmo** [comment]
I addressed the spacing consistency issues and adjusted a few indenting issues that I encountered as well

File: test/powershell/Modules/Microsoft.PowerShell.Management/Get-ComputerInfo.Tests.ps1 [hunk 6]
```diff
@@ -0,0 +1,1367 @@
...
+    {
+$signature = @"
+[DllImport("kernel32.dll")]
+public static extern bool GetFirmwareType(ref uint firmwareType);
+"@
+        Add-Type -MemberDefinition $signature -Name "Win32BiosFirmwareType" -Namespace Win32Functions -PassThru
+    }
+
+    function Get-PhysicallyInstalledSystemMemory
+    {
```
Comment (ChrisUbben) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Get-ComputerInfo.Tests.ps1]: Everything is artificially indented after this point. Remove the indent if not nested.

**Comment by ChrisUbben** [comment]
LGTM

**Pull request merged**